### PR TITLE
feat: add C implementation for `stats/base/dists/bernoulli/entropy`

### LINKED ISSUE #3386: [RFC]: Add C implementation for `@stdlib/stats/base/dists/arcsine/logcdf`
### Description

This RFC proposes adding a C implementation, including native Node.js addon, C benchmarks and C examples, for the `@stdlib/stats/base/dists/arcsine/logcdf` package.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/arcsine/logcdf" make install-node-addons
```

#### Run C Examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logcdf/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logcdf/benchmark/c/benchmark.c"
```

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### LINKED ISSUE #3412: [RFC]: Add C implementation for `@stdlib/stats/base/dists/bernoulli/entropy`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/bernoulli/entropy`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/bernoulli/entropy" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/bernoulli/entropy/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vivekmaurya001**
Resolves #3412.

## Description

- adds C implementation for @stdlib/stats/base/dists/bernoulli/entropy along with relevant tests, benchmarks and examples

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/3412
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md [hunk 1]
```diff
@@ -130,6 +130,93 @@ for ( i = 0; i < 10; i++ ) {
...
+Returns the [entropy][entropy] of a [Bernoulli][bernoulli-distribution] distribution with success probability `p` (in [nats][nats]).
+
+```c
+double out = stdlib_base_dists_bernoulli_entropy( 0.1 );
+// returns ~0.325
+```
+
+The function accepts the following arguments:
+
+-   **x**: `[in] double` minimum support
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md]: You're missing periods in your descriptions.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md [hunk 2]
```diff
@@ -130,6 +130,93 @@ for ( i = 0; i < 10; i++ ) {
...
+#include "stdlib/stats/base/dists/bernoulli/entropy.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main( void ) {
+    double p;
+    double y;
+    int i;
+    
+    for ( i = 0; i < 25; i++ ) {
+		p = ( (double)rand() / (double)RAND_MAX ) ;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md]: Mixed tabs and spaces. See `arcsine/cdf` for what we expect here in terms of style.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/Makefile [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#/
+# Removes generated files.
+#
+# @example
+# make clean
+#/
+clean:
+	$(QUIET) -rm -f *.o *.out
+
+.PHONY: clean
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/Makefile]: Missing trailing newline. Please make sure you have setup your local dev environment correctly, including EditorConfig.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+static double benchmark( void ) {
+	double elapsed;
+	double p;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		p = ( (double)rand() / (double)RAND_MAX ) ;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/benchmark.c]: See `arcsine/cdf` for what we expect.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
+		elapsed = benchmark();
+		print_results( elapsed );
+		printf( "ok %d benchmark finished\n", i+1 );
+	}
+	print_summary( REPEATS, REPEATS );
+}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/benchmark.c]: Missing trailing newline.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/binding.gyp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+          'action': [
+            'cp',
+            '<(PRODUCT_DIR)/<(addon_target_name).node',
+            '<(addon_output_dir)/<(addon_target_name).node',
+          ],
+        },
+      ], # end actions
+    }, # end target copy_addon
+  ], # end targets
+}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/binding.gyp]: Missing trailing newline.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/src/main.c [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+/**
+* Returns the entropy of a Bernoulli distribution.
+*
+* @param p    success probability
+* @return     entropy
+*
+* @example
+* double y = stdlib_base_dists_bernoulli_entropy( 0.1 );
+* // returns ~0.325
+* @example
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/src/main.c:33]:
```suggestion
*
* @example
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/test/test.native.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+var entropy = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( entropy instanceof Error )
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/test/test.native.js]:
```suggestion
tape( 'main export is a function', opts, function test( t ) {
```

This applies here and elsewhere in this file.

**Review by kgryte — CHANGES_REQUESTED** [review]
@vivekmaurya001 Before opening many simultaneous PRs, please prioritize improving your existing PRs. Many of the mistakes flagged in this PR will need to be corrected across your other PRs. You'll save yourself (and us!) work and time by getting things right the first time.

**Comment by vivekmaurya001** [comment]
@kgryte , I understand my mistake , I am correcting it

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md (hunk 1 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md]: done this

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/test/test.native.js (hunk 8 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/test/test.native.js]: done

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/src/main.c (hunk 7 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/src/main.c:33]: done

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md (hunk 2 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md]: done

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/Makefile (hunk 3 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/benchmark/c/Makefile]: done this

**Comment by vivekmaurya001** [comment]
@kgryte , changes done please give a look !

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/manifest.json [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        ],
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/assert/is-nan",
+          "@stdlib/math/base/special/ln"
+        ]
+      }
+    ]
+  }
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/manifest.json:79]:
```suggestion
  }

```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md [hunk 10]
```diff
@@ -130,6 +130,92 @@ for ( i = 0; i < 10; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/stats/base/dists/bernoulli/entropy.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main( void ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/entropy/README.md]:
```suggestion
#include "stdlib/stats/base/dists/bernoulli/entropy.h"
#include <stdlib.h>
#include <stdio.h>

int main( void ) {
```

**Review by Planeshifter — APPROVED** [review]
Let's get this one finally in...

**Pull request merged**